Druid/Exception.extension.st

```diff
@@ -0,0 +1,9 @@
+Extension { #name : #Exception }
+
+{ #category : #'*Druid' }
+Exception >> handlePrimitiveCompilationFor: aDRPrimitiveCCCollector [
+
+	aDRPrimitiveCCCollector
+		failedPrimitivesAt: self
+		add: aDRPrimitiveCCCollector sourceName
```

Comment:
Is the extension method really justified here? There is no polymorphism, nor access to the exception state, right ?

Reply:
I don't understand why there should be polymorphism or access to the exception state?

Reply:
Because if not why is this method here :P

Reply:
This technique allows different exceptions to respond to primitive compilation errors in different ways. A more generic collector (like this one) can simply gather all the exceptions in a Dictionary, so that they can be inspected later without having to suffer the constant opening of a Debugger on each compilation error. Due to the stage of the project this happens quite often if one wants to implement a tool for a minimal inspection of the implementation (outside the tests which are a safe environment and where one is sure that everything will work as expected). This logic is implemented in several places in the image (e.g. #manageTestProcessBy:, #notifyUserOfCommand:, #recordResultOf: inHDTestReport:, #stDebuggerInspectorNodesFor:). It also allows it to be re-implemented more generically for any type of "AbstractMethodCompilerCompiler", e.g. DRMethodCompilerCompiler, DRBytecodeCompilerCompiler.

One thing I can do is to move this class to a ToolsSupport package and thus not "mess up" the current implementation.